src/rf/apps/workers/process.py

```diff
@@ -221,7 +227,38 @@ def _check_timeout(self, record):
 
         return True
 
-    def _save_result(self, record):
+    def _copy_image(self, record):
+        """
+        Copy an image into the S3 bucket from an external source.
+        data -- attribute data from SQS.
+        """
+
+        try:
+            data = record['data']
+            image_id = data['image_id']
+        except KeyError:
+            return False
+
+        try:
+            int(image_id)
+        except ValueError:
+            return False
+
+        try:
+            image = LayerImage.objects.get(id=image_id)
+        except LayerImage.DoesNotExist:
+            return False
+
+        if image.source_s3_bucket_key:
+            success = s3_copy(image.source_s3_bucket_key, image.get_s3_key())
+            if not success:
+                status_updates.mark_image_invalid(
+                    image.s3_uuid, ERROR_MESSAGE_IMAGE_TRANSFER)
+            return True
```

Comment:
The pattern I've been using elsewhere is to return the value of the mark_invalid call. The update status functions should return a boolean True if they saved properly. That way if something went wrong, we leave the message on the queue and it will be tried again.